multiqc/modules/bases2fastq/bases2fastq.py

```diff
@@ -0,0 +1,321 @@
+        if len(self.b2f_data) == 0:
+            log.error("No Samples are found.")
+            log.error("Please visit Elembio docs for more information - https://docs.elembio.io/docs/bases2fastq/")
+            raise UserWarning
+        log.info(f"Found {num_samples} samples within bases2fastq results, and {len(self.b2f_data)} samples")
+
+        # Group by run name
+        self.group_dict = OrderedDict()
+        self.group_lookup_dict = dict()
+        for s_name in self.b2f_data.keys():
+            s_group = self.b2f_data[s_name]["RunName"]
+
+            if not self.group_dict.get(s_group):
+                self.group_dict.update({s_group: []})
+
+            self.group_dict[s_group].append(s_name)
+            self.group_lookup_dict.update({s_name: s_group})
+
+        # Assign project
+        for s_name in self.b2f_data.keys():
+            if project_lookup_dict.get(s_name):
+                s_group = project_lookup_dict[s_name]
+                if not self.group_dict.get(s_group):
+                    self.group_dict.update({s_group: []})
+                self.group_dict[s_group].append(s_name)
+                self.group_lookup_dict.update({s_name: s_group})
+
+        # Assign color for each group
+        n_colors = len(self.group_dict.keys())
+        """
+        palette = [
+            "rgba({r},{g},{b},0.5)".format(r=rgb[0] * 255, g=rgb[1] * 255, b=rgb[2] * 255)
+            for rgb in sns.color_palette("bright", n_colors)
+        ]
+        """
+        self.color_getter = mqc_colour.mqc_colour_scale()
+        self.palette = sum(
+            [
+                self.color_getter.get_colours(hue)
+                for hue in ["Set2", "Pastel1", "Accent", "Set1", "Set3", "Dark2", "Paired", "Pastel2"]
+            ],
+            [],
+        )
+        if len(self.group_dict) > len(self.palette):
+            hex_range = 2**24
+            extra_colors = [hex(random.randrange(0, hex_range)) for _ in range(len(self.group_dict), len(palette))]
+            self.palette = self.palette + extra_colors
+        self.group_color = {g: c for g, c in zip(self.group_dict.keys(), self.palette[: len(self.group_dict)])}
+        self.sample_color = dict()
+        for s_name in self.b2f_data.keys():
+            self.sample_color.update({s_name: self.group_color[self.group_lookup_dict[s_name]]})
+        self.run_color = copy.deepcopy(self.group_color)  # Make sure that run colors and group colors match
+        self.palette = self.palette[len(self.group_dict) :]
+
+        # Read custom group info
+        self.group_info_exist = False
+        for f in self.find_log_files("bases2fastq/group"):
+            if self.group_info_exist:
+                log.warning(
+                    "More than one group assignment files are found. Please only keep one assignment file in the analysis folder. Bases2fastq stats will not be plotted"
+                )
+            group_info = pd.read_csv(StringIO(f["f"]))
```

Comment:
Pandas is not imported anywhere.

Note that we also don't install pandas with MultiQC, so we can't really use it. I think it's pretty straightforward to get around without pandas here?

Reply:
Modified it in the new :)